src/main/scala/scala/xml/parsing/MarkupParserCommon.scala

```diff
@@ -247,7 +247,7 @@ private[scala] trait MarkupParserCommon extends TokenTests {
       while (true) {
         if (ch == head && peek(rest))
           return handler(positioner(), sb.toString)
-        else if (ch == SU)
+        else if (ch == SU || eof)
           truncatedError("") // throws TruncatedXMLControl in compiler
 
```

Comment:
@shimamoto Ok, cool! I thought that was odd.  Thanks for fixing up the unit tests!